app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -29,6 +29,7 @@ def find_discovery_rule host
 
   # trigger the provisioning
   def perform_auto_provision host, rule
+    host = host.becomes(::Host::Managed)
```

Comment:
Is this really needed? There were some hidden dragons and if we can work with the generic instance, I'd rather stick with it.